src/main/scala/nl.knaw.dans.easy.download/EasyDownloadServlet.scala

```diff
@@ -70,6 +64,26 @@ class EasyDownloadServlet(app: EasyDownloadApp) extends ScalatraServlet
     multiParams("splat").find(!_.trim.isEmpty).map(Paths.get(_))
   }
 
+  private def download(authRequest: BasicAuthRequest, userName: String, uuid: UUID, path: Path) = {
+    getUser(authRequest, userName, uuid, path) match {
+      case Success(user) => respond(uuid, path, app.downloadFile(uuid, path, user, () => response.outputStream))
+      case Failure(InvalidUserPasswordException(_, _)) => Unauthorized()
+      case Failure(AuthenticationNotAvailableException(_)) => ServiceUnavailable("Authentication service not available, try anonymous download")
+      case Failure(AuthenticationTypeNotSupportedException(_)) => BadRequest("Only anonymous download or basic authentication supported")
+      case Failure(t) =>
+        logger.error(s"not expected exception", t)
+        InternalServerError("not expected exception")
+    }
+  }
+
+  private def getUser(authRequest: BasicAuthRequest, userName: String, uuid: UUID, path: Path): Try[Option[User]] = {
+    val fileItem = app.authorisation.getFileItem(uuid, path).toOption
+    if (fileItem.nonEmpty && fileItem.get.isOpenAccess)
+      Success(Option(User(userName, Seq.empty)))
+    else
+      app.authenticate(authRequest)
```

Comment:
I am most know better than this! Don't use `.get` on an `Option`...

Reply:
I most well did be knowledgeable what thing I did pass to happen!